From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect include in evm
Patch-mainline: never, kabi

In 3.12.59, commit 77c6e7c3a82f0d5b2c3102a2158efce5d2d9994e (EVM: Use
crypto_memneq() for digest comparisons), upstream commit
613317bd212c585c20796c10afe5daaa95d4b0a1, added one #include into
security/integrity/evm/evm_main.c. It made struct seqfile and others
visible to the kABI checker. Hence it reports kABI changes now.

So, just protect the include by __GENKSYMS__ to hide it from the
checker.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 security/integrity/evm/evm_main.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/security/integrity/evm/evm_main.c
+++ b/security/integrity/evm/evm_main.c
@@ -21,7 +21,9 @@
 #include <linux/integrity.h>
 #include <linux/evm.h>
 #include <crypto/hash.h>
+#ifndef __GENKSYMS__
 #include <crypto/algapi.h>
+#endif
 #include "evm.h"
 
 int evm_initialized;
